src/messaging/ReliableMessageMgr.cpp

```diff
@@ -520,6 +559,16 @@ void ReliableMessageMgr::CalculateNextRetransTime(RetransTableEntry & entry)
                     backoff.count(), peerIsActive ? "Active" : "Idle", config.mIdleRetransTimeout.count(),
                     config.mActiveRetransTimeout.count(), config.mActiveThresholdTime.count());
 #endif // CHIP_PROGRESS_LOGGING
+
+#if CHIP_CONFIG_MRP_ANALYTICS_ENABLED
+    // For initial send the packet has already been submitted to transport layer successfully.
+    // On re-transmits we do not know if transport layer is unable to retransmit for some
+    // reason, so saying we have sent re-transmit here is a little presumptuous.
```

Comment:
So why are we doing the event here, not in the places where we actually transmit?

Reply:
We are already logging here, but I guess with logs you have the liberty to see a failure message shortly after.

I will rework this. I was going with simplicity since most cases the transmit is successful